cmd/kwil-admin/cmds/snapshot/create.go

```diff
@@ -0,0 +1,253 @@
+package snapshot
+
+import (
+	"bufio"
+	"bytes"
+	"compress/gzip"
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"math/big"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/kwilteam/kwil-db/cmd/kwild/config"
+
+	"github.com/spf13/cobra"
+)
+
+var (
+	createLongExplain = `This command creates a logical SQL dump of the specified database and it's snapshot hash in the directory specified. This command is intended for generating exportable snapshots that are not integrated into the node's snapshot store on the running KwilDB instance. Instead, they are meant to be used for initializing a new network, serving as the genesis state. 
+	
+	This command interacts directly with the Kwild's PostgreSQL server, bypassing any interactions with the kwild node. It requires user with administrative privileges on the database to ensure that it can access all necessary data and perform required actions such as locking tables etc. without restrictions that are beyond the permissions of a standard database user.`
+
+	createExample = `# Create snapshot and snapshot hash of the database
+kwil-admin snapshot create kwil-db kwil_user localhost 5432 /path/to/snapshot/dir
+
+# Snapshot and hash files will be created in the snapshot directory
+ls /path/to/snapshot/dir
+kwildb-snapshot-hash    kwildb-snapshot.sql.gz`
+
+	createDatabase = `CREATE DATABASE `
+)
+
+/*
+Use this at the beginning of the sql dump file to drop any active connections on the 'kwild' database.
+Not a good idea to use this if the Kwild node is connected to this database.
+
+SELECT pg_terminate_backend(pg_stat_activity.pid)
+FROM pg_stat_activity
+WHERE pg_stat_activity.datname = 'kwild'
+  AND pid <> pg_backend_pid();
+*/
+
+func createCmd() *cobra.Command {
+	var snapshotDir, dbName, dbUser, dbPass, dbHost, dbPort string
+	cmd := &cobra.Command{
+		Use:     "create",
+		Short:   "Creates a snapshot of the database.",
+		Long:    createLongExplain,
+		Example: createExample,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			snapshotDir, err := expandPath(snapshotDir)
+			if err != nil {
+				return fmt.Errorf("failed to expand snapshot directory path: %w", err)
+			}
+
+			return pgDump(cmd.Context(), dbName, dbUser, dbPass, dbHost, dbPort, snapshotDir)
+		},
+	}
+
+	cmd.Flags().StringVar(&snapshotDir, "snapdir", "kwild-snaps", "Directory to store the snapshot and hash files")
+	cmd.Flags().StringVar(&dbName, "dbname", "kwild", "Name of the database to snapshot")
+	cmd.Flags().StringVar(&dbUser, "user", "postgres", "User with administrative privileges on the database")
+	cmd.Flags().StringVar(&dbPass, "pass", "", "Password for the database user")
```

Comment:
I believe Postgres's default password is password, right?

Also, could you make the full flag `--password`?